lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -3370,6 +3441,30 @@ bool LLVMToSPIRVBase::transGlobalVariables() {
   for (auto I = M->global_begin(), E = M->global_end(); I != E; ++I) {
     if ((*I).getName() == "llvm.global.annotations")
       transGlobalAnnotation(&(*I));
+    else if ([&]() -> bool {
+               // Check if the GV is used only in var/ptr instructions. If yes -
+               // skip processing of this since it's only an annotation GV.
+               bool Return = false;
+               for (auto *U : I->users()) {
+                 auto *V = U->stripPointerCasts();
+                 if (GetElementPtrInst *GEP =
+                         dyn_cast_or_null<GetElementPtrInst>(V))
+                   V = GEP;
+                 if (IntrinsicInst *II = dyn_cast_or_null<IntrinsicInst>(V)) {
+                   switch (II->getIntrinsicID()) {
+                   case Intrinsic::var_annotation:
+                   case Intrinsic::ptr_annotation:
+                     Return = true;
+                     break;
+                   default:
+                     return false;
+                   }
+                 } else
+                   return false;
+               }
+               return Return;
+             }())
```

Comment:
```suggestion
               // Check if the GV is used only in var/ptr instructions. If yes -
               // skip processing of this since it's only an annotation GV.
               for (auto *U : I->users()) {
                 Value *V = U->stripPointerCasts();
                 if (auto *GEP =
                         dyn_cast_or_null<GetElementPtrInst>(V))
                   V = GEP;
                 if (auto *II = dyn_cast_or_null<IntrinsicInst>(V)) {
                   switch (II->getIntrinsicID()) {
                   case Intrinsic::var_annotation:
                   case Intrinsic::ptr_annotation:
                     continue;
                   default:
                     return false;
                   }
                 }
                 return false;
               }
               return true;
             }())
```

Reply:
Ouch, I actually need to apply logic from [here](https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/1191#discussion_r699990957) first, then check for `user_empty` and only then apply this comment :)

Reply:
And actually I had to go through GEP users as well.